src/watcher.js

```diff
@@ -115,41 +119,174 @@ async function getLastBlockToProcess() {
 
-    const fromBlock = lastProcessedBlock.add(ONE)
-    const toBlock = lastBlockToProcess
+          const events = await getEvents({
+            contract: eventContract,
+            event: config.event,
+            fromBlock,
+            toBlock,
+            filter: config.eventFilter
+          })
+          logger.info(
+            `Found ${events.length} ${config.event} events for contract address
+              ${eventContract.options.address}`
+          )
 
-    const events = await getEvents({
-      contract: eventContract,
-      event: config.event,
-      fromBlock,
-      toBlock,
-      filter: config.eventFilter
-    })
-    logger.info(`Found ${events.length} ${config.event} events`)
+          await processBridgeMappingsAdded(events)
 
-    if (events.length) {
-      const job = await processEvents(events)
-      logger.info('Transactions to send:', job.length)
+          logger.debug(
+            { lastProcessedBlock: lastBlockToProcess.toString() },
+            'Updating last processed block'
+          )
+          await updateLastProcessedBlock(lastBlockRedisKey, lastBlockToProcess)
+        }
+      } else {
+        const mapper = async bridgeObj => {
+          const homeStartBlock = toBN(bridgeObj.homeStartBlock || 0)
+          const foreignStartBlock = toBN(bridgeObj.foreignStartBlock || 0)
+          const bridgeContractAddress =
+            config.id === 'erc-erc-multiple-affirmation-request'
+              ? bridgeObj.foreignBridge
+              : bridgeObj.homeBridge
+          const eventContractAddress =
+            config.id === 'erc-erc-multiple-affirmation-request'
+              ? bridgeObj.foreignToken
+              : bridgeContractAddress
+          const eventFilter =
+            config.id === 'erc-erc-multiple-affirmation-request'
+              ? { to: bridgeObj.foreignBridge }
+              : {}
+          const bridgePair = `${bridgeObj.homeBridge}:${bridgeObj.foreignBridge}`
+          const lastBlockRedisKey = `${lastBlockRedisKeyDefault}:${bridgePair}`
+          const lastProcessedBlock = await getLastProcessedBlock(
+            lastBlockRedisKey,
+            config.id === 'erc-erc-multiple-affirmation-request'
+              ? BN.max(foreignStartBlock.sub(ONE), ZERO)
+              : BN.max(homeStartBlock.sub(ONE), ZERO)
+          )
+          const lastBlockToProcess = await processOne(
+            sendToQueue,
+            bridgeContractAddress,
+            eventContractAddress,
+            bridgeObj.homeBridge,
+            bridgeObj.foreignBridge,
+            eventFilter,
+            lastProcessedBlock
+          )
+          return { lastBlockToProcess, lastBlockRedisKey }
+        }
 
-      if (job.length) {
-        await sendToQueue(job)
+        const deployedBridges = await getDeployedBridges()
+        logger.debug(`Found ${deployedBridges.length} deployed bridges`)
+        const results = await pMap(deployedBridges, mapper, { concurrency })
```

Comment:
I think we can achieve the same behavior using `promise-limit` that is already a dependency of the project.

We could define a method on `src/utils/utils.js` as follow:
```javascript
const promiseLimit = require('promise-limit')

...

async function processConcurrently(array, f, concurrency) {
  const limit = promiseLimit(concurrency)
  const promises = await array.map(obj => limit(async () => f(obj)))
  return Promise.all(promises)
}
```

And then update this line to:
```javascript
const results = await processConcurrently(deployedBridges, mapper, concurrency)
```

Does it makes sense?